qiskit/opflow/expectations/aer_pauli_expectation.py

```diff
@@ -93,8 +94,6 @@ def _replace_pauli_sums(cls, operator):
             snapshot_instruction = SnapshotExpectationValue("expval_measurement", paulis)
             snapshot_op = CircuitStateFn(snapshot_instruction, is_measurement=True)
             return snapshot_op
-        if isinstance(operator, ListOp):
```

Comment:
If we remove this, will it still work for objects that derive from list ops for which `operator.__class__ == ListOp` fails (like e.g. `ComposedOp`)?

Reply:
that case is not supported originally. 
```python
converted = AerPauliExpectation().convert(StateFn(ComposedOp([X, Z]), is_measurement=True) @ Zero)
CircuitSampler(Aer.get_backend("qasm_simulator")).convert(converted)
```
main
```
ValueError: Composition with a Statefunction in the first operand is not defined.
```
this fix
```
TypeError: unsupported operand type(s) for *: 'NoneType' and 'float'
```

In the original case, the conversion can be done, but the conversion result is not correct (undefined behavior).
I think that an error should be raised in early stage.